### PR TITLE
Add Subscription BloC and HomePage

### PR DESCRIPTION
**Pull request by satyamurti**
- Created Companies Bloc for Streaming of Stock Prices
- Added Responsiveness using LayoutBuilder
- Added Subscribe and Unsubscribe Cubit for handling subscription of rpc streams
- Added GetStockList Bloc to get initial List of Stocks
- Created basic UI for Home Page accordingly 
- Commented out Dalal App Icon Image from landing page because auth and landing pages are not responsive , they are overflown in many screens.

### CODE REVIEW

File: lib/blocs/user/user_state.dart [hunk 1]
```diff
@@ -43,3 +43,19 @@ class UserDataLoaded extends UserState {
   @override
   List<Object> get props => [sessionId, user];
 }
+
+class SubscriptionDataLoaded extends UserState {
```
Comment (Sudhindra3000) [inline lib/blocs/user/user_state.dart:47]: `UserBloc` is only meant for User authentication state. How about we shift this logic to a separate `SubscriptionBloc`?

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/blocs/user/user_state.dart (hunk 1 above)
Comment (Sudhindra3000) [reply lib/blocs/user/user_state.dart:47]: And `SubscriptionBloc` should be only accesible to the Home Pages. We need to see how to implement that

Comment (Sudhindra3000) [reply lib/blocs/user/user_state.dart:47]: Or we could even have all the subscription ids in User data itself. In splash page when we are getting user data from stored session id, we can make all the common subscription calls, and store the ids in `UserDataLoaded` state

Comment (siva2204) [reply lib/blocs/user/user_state.dart:47]: And we have  to call `unSubscribe` with the subscription I'd, once we're closing with stream.

**Review by dhruv1294 — APPROVED** [review]

**Comment by dhruv1294** [comment]
btw is resposive a typo??

**Comment by satyamurti** [comment]
> btw is resposive a typo??

Yes sorry my bad

File: lib/pages/home_page.dart [hunk 2]
```diff
@@ -1,39 +1,330 @@
...
+  void dispose() {
+    context.read<SubscribeCubit>().unsubscribe(_stockPricesSubscriptionId!);
+    super.dispose();
+  }
+
+  @override
+  initState() {
+    super.initState();
+    // Get List of Stocks
+    BlocProvider.of<CompaniesBloc>(context).add(const GetStockList());
```
Comment (Sudhindra3000) [inline lib/pages/home_page.dart:37]: use `context.read()` here too instead of `BlocProvider.of()` ?

File: lib/pages/home_page.dart [hunk 3]
```diff
@@ -1,39 +1,330 @@
...
-              children: [
-                Text(
-                  'Current User: ${user.name}\n Session Id: ${getIt<String>()}',
-                  style: const TextStyle(fontSize: 18),
-                  textAlign: TextAlign.center,
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  SubscriptionId? _stockPricesSubscriptionId;
```
Comment (Sudhindra3000) [inline lib/pages/home_page.dart:25]: Remove this field from the Widget and integrate in the Bloc itself?

Comment (satyamurti) [reply lib/pages/home_page.dart:25]: but for unsubscribing we need to pass this id and we call this in dispose function of widget

Comment (Sudhindra3000) [reply lib/pages/home_page.dart:25]: The subId is there in `SubscriptionDataLoaded` state right. In dispose function we can get current state of subBloc with subBloc.state, cast it to `SubscriptionDataLoaded`to get the subId.

Comment (satyamurti) [reply lib/pages/home_page.dart]: Okay I did that

File: lib/pages/home_page.dart (hunk 2 above)
Comment (satyamurti) [reply lib/pages/home_page.dart]: Okay

File: lib/pages/home_page.dart [hunk 4]
```diff
@@ -1,39 +1,345 @@
...
+
+class _HomePageState extends State<HomePage> {
+  // Unsubscribe to the streams when the widget is disposed
+  @override
+  void dispose() {
+    SubscriptionId? _stockPricesSubscriptionId;
+    final state = context.read<SubscribeCubit>().state;
+    if (state is SubscriptionDataLoaded) {
+      _stockPricesSubscriptionId = state.subscriptionId;
+      context.read<SubscribeCubit>().unsubscribe(_stockPricesSubscriptionId);
```
Comment (Sudhindra3000) [inline lib/pages/home_page.dart:32]: remove the local field `SubscriptionId? _stockPricesSubscriptionId` and just do `context.read<SubscribeCubit>().unsubscribe()`. We can do the casting `state is SubscriptionDataLoaded` directly inside the unsubscribe function

File: lib/pages/home_page.dart [hunk 5]
```diff
@@ -1,39 +1,345 @@
...
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        const Text(
+                          'Home',
+                          style: TextStyle(
+                              fontSize: 18,
+                              fontWeight: FontWeight.w500,
+                              color: whiteWithOpacity75),
+                          textAlign: TextAlign.center,
+                        ),
```
Comment (Sudhindra3000) [inline lib/pages/home_page.dart:79]: The widget tree is pretty big. Split it into sub parts with meaningful names. Refer `LoginPage` and `RegisterPage` once

**Review by Sudhindra3000 — APPROVED** [review]

**Pull request merged**